Objects/tupleobject.c

```diff
@@ -391,11 +385,8 @@ _PyTuple_FromArray(PyObject *const *src, Py_ssize_t n)
 }
 
 PyObject *
-_PyTuple_FromArraySteal(PyObject *const *src, Py_ssize_t n)
+_PyTuple_FromNonEmptyArraySteal(PyObject *const *src, Py_ssize_t n)
```

Comment:
I don't see how this change is related to the fix. And I don't see the benefits to reject n=0.

Reply:
I've reverted it.

`_PyTuple_FromArraySteal` is only called from the interpreter so we know that `n != 0`. I can change it in another PR.